### PR TITLE
Allow editing of tags

### LINKED ISSUE #69: Edit the tag the message belongs to
When editing a message I should be able to either:

* Change the tag the message belongs to
* Set the tag for the message, if it is untagged.

### PR DESCRIPTION
**Pull request by jonnhy12333**
So this one was pretty simple, just added the primaryTag to the edit message dialog:
image

I put it above the message because it comes before the message when we display it in the chat form?  I did set focus to the 'Message' part of this dialog when you launch because I'm assuming that users probs want to edit the message the majority of the time.

 - The tag is obviously not a required field and can be left empty / creating an untagged message.
 - Max length on the tag field is 20 characters (i know there are other display issues with long tags, but that's a different story)
 - Fixes #69
 - Tag preview in top right of dialog (tag can be cleared by clicking the 'x' on that chip as well)
 - Tags are now encrypted in db

### CODE REVIEW

**Comment by greggiannone** [comment]
Wait so this allows for tags to have spaces? That seems like a problem cuz you can't type tags normally like that

**Comment by jonnhy12333** [comment]
> Wait so this allows for tags to have spaces? That seems like a problem cuz you can't type tags normally like that

Oooo good catch... just added validator for the tag name to prevent spaces (or any whitespace).  It allows letters, numbers, hyphens, or underscores

File: src/app/directives/tag-validator.directive.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    if (control.value) {
+      if (/\s/.test(control.value)) {
+        return {
+          tag: 'Tag must not contain white space',
+        };
+      }
+      if (!/^([a-zA-Z0-9_\-\+]+)$/.test(control.value)) {
+        return {
+          tag: 'Tag must only consist of letters, numbers, hyphens, and underscores',
```
Comment (greggiannone) [inline src/app/directives/tag-validator.directive.ts]: Can't the whitespace bit be covered by this as well?

Comment (jonnhy12333) [reply src/app/directives/tag-validator.directive.ts]: yep, whoops, reduced to one regular expression now

**Pull request merged**